ui/src/app/i18n.js

```diff
@@ -0,0 +1,11 @@
+import i18n from 'i18next';
+import Backend from 'i18next-http-backend';
+import LanguageDetector from 'i18next-browser-languagedetector';
+import {initReactI18next} from 'react-i18next';
+
+i18n.use(Backend).use(LanguageDetector).use(initReactI18next).init({
+    fallbackLng: 'en',
+    debug: false
+});
```

Comment:
two problems I see with this approach:

1. There's no way to switch languages in the UI right now. You're forced into using the language your browser detects for your locale, even if you don't like or don't prefer that translation.
1. Translations are currently hosted in core in this approach, but core maintainers may not be able to maintain these translations (per https://github.com/argoproj/argo-workflows/pull/12682#pullrequestreview-1889637846 and https://github.com/argoproj/argo-workflows/issues/10103#issuecomment-1327717345).
    - I think a plugin based approach, similar to the Jenkins Chinese plugin that you maintain as you mentioned in [the issue](https://github.com/argoproj/argo-workflows/issues/10103) would be easier to maintain and be more explicit about the level of support (since translations would be done by contributors). We could host translations in `argoproj-labs` too if desired.
        - How to implement the plugin based approach is a bit more complicated as UI plugins don't yet exist (see also #6943. The existing UI subtask #6945 also does not outline an approach for "UI extensions" such as this that could modify the entire UI, as opposed to just add a new page).
           - A very simple, naive approach would be to allow users to add a URL to a translation file that is fetched. Then we can store that in `localStorage`
           - Another approach would be for the API to serve translation files located in a specific directory on disk. Then one can add a volume with those or make a custom image with them.
               - This roughly follows how [Argo CD loads UI extensions](https://argo-cd.readthedocs.io/en/stable/developer-guide/extensions/ui-extensions/) and is more secure as well (it requires an operator to add the config and is served internally instead of externally, the latter of which could violate the CSP if not properly configured)
            - The other common approach, particularly in corporate settings, is to host translations on a separate website with a translation CMS of sorts. There are OSS tools like [Weblate](https://github.com/WeblateOrg/weblate) and others that are also often free to use for other OSS projects.
              - This is probably the most robust and translation-specific approach

Reply:
I can add a drop-down menu for language switch.

Having a localization plugin will make it be easier to maintain and publish. Currently I am able to help on the plugin. A very important thing is that the particular language (Chinese for this one) committer is necessary. There is no one willing to do that for the Chinese localization, I prefer to pause this PR.

Reply:
> A very important thing is that the particular language (Chinese for this one) committer is necessary. There is no one willing to do that for the Chinese localization, I prefer to pause this PR.

Yes that's exactly what I meant! I'm not sure if we have anyone able & willing to maintain it actively.

And we definitely couldn't do that for all localizations, so community plugins would make more sense _overall_ as an approach for _all_ localizations.

For Chinese, for instance, you would be able to maintain that specific localization in its own repo (or in a shared `argo-workflows-ui-localizations` repo or something, split by `CODEOWNERS` similar to `argo-helm` for instance).
We could also potentially package some more maintained ones into the UI by default (with the caveat that the default ones could quickly go out-of-date).
If we were to use a hosted site, we could add contributors as editors for different localizations.